cmake/Modules/FindMKLDNN.cmake

```diff
@@ -39,6 +39,9 @@ IF(MKL_FOUND)
   # Append to mkldnn dependencies
   LIST(APPEND MKLDNN_LIBRARIES ${MKL_LIBRARIES})
   LIST(APPEND MKLDNN_INCLUDE_DIR ${MKL_INCLUDE_DIR})
+  IF(MKLDNN_USE_CBLAS)
+    ADD_DEFINITIONS(-DUSE_MKL -DUSE_CBLAS)
```

Comment:
Here the change looks quite generic (`USE_CBLAS`). Are you sure it won't affect other modules? Do you think it might be better to make the change [here](https://github.com/pytorch/pytorch/blob/fd6e11123adb71864ea3343abc382e69c1f4593b/cmake/Modules/FindMKLDNN.cmake#L106)?

Reply:
In Pytorch/Caffe2, only the third_party/ideep/mkldnn use the “USE_CBLAS” to decide whether we need to use blas algorithms for computations or not. We add the USE_MKL and USE_CBLAS build options here so that we can decide whether use CBLAS in MKL-DNN from Pytorch/Caffe2 building side. 
![image](https://user-images.githubusercontent.com/35086899/60313983-99bc2f80-9993-11e9-8a18-1e79997c1432.png)

Reply:
Change the add_definition to TARGET_COMPILE_DEFINITIONS. See the following answer. Thanks.